File: MusicDemoOC/MusicDemo/Controller/ViewController.m
//
//  ViewController.m
//  MusicDemo
//
//  Created by 沈凯 on 2017/11/7.
//  Copyright © 2017年 Ssky. All rights reserved.
//

#import "ViewController.h"
#import "MusicModel.h"
#import "MainPageCell.h"
#import "DetailViewController.h"

@interface ViewController ()<UITableViewDelegate,UITableViewDataSource>
@property (weak, nonatomic) IBOutlet UITableView *tableView;
@property (strong, nonatomic) NSMutableArray *data;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
//    数据加载
    [self loadData];
//    注册Cell
    [self.tableView registerNib:[UINib nibWithNibName:@"MainPageCell" bundle:[NSBundle mainBundle]] forCellReuseIdentifier:@"MainPageCell"];
}
#pragma mark 数据加载
- (void)loadData
{
    NSMutableArray *arr = [NSMutableArray arrayWithContentsOfFile:[[NSBundle mainBundle]pathForResource:@"Musics" ofType:@"plist"]];
    for (NSDictionary *dic in arr) {
        MusicModel *model = [[MusicModel alloc]initWithDictionary:dic];
        [self.data addObject:model];
    }
}
#pragma mark 懒加载数组
- (NSMutableArray *)data
{
    if (!_data) {
        _data = [NSMutableArray array];
    }
    return _data;
}
#pragma mark - TableViewDelegate
- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    return 1;
}
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return self.data.count;
}
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    MainPageCell *cell = [tableView dequeueReusableCellWithIdentifier:@"MainPageCell"];
    cell.model = self.data[indexPath.row];
    
    return cell;
}
- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
{
    return 118;
}
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    DetailViewController *vc = [DetailViewController new];
    vc.model = self.data[indexPath.row];
    vc.data = self.data;
    vc.currentIndex = indexPath.row;
    dispatch_async(dispatch_get_main_queue(), ^{
        [self presentViewController:vc animated:YES completion:nil];
    });
}
@end

File: MusicDemoSwift/MusicDemoSwift/Controller/ViewController.swift
//
//  ViewController.swift
//  MusicDemoSwift
//
//  Created by 沈凯 on 2017/11/13.
//  Copyright © 2017年 Ssky. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
//    TableView
    @IBOutlet weak var tableView: UITableView!
//    懒加载数组
    lazy var data = [MusicModel]()
    
    override func viewDidLoad() {
        super.viewDidLoad()
//        数据加载
        loadData()
//        注册Cell
        tableView.register(UINib(nibName: "MainPageCell", bundle: Bundle.main), forCellReuseIdentifier: "MainPageCell")
    }
//    MARK: - 数据加载
    func loadData() {
        let arr = NSMutableArray.init(contentsOfFile: Bundle.main.path(forResource: "Musics", ofType: "plist")!)
        for dic in arr! {
            let model = MusicModel.init(dic: dic as! [String : AnyObject])
            data.append(model)
        }
    }
}
//MARK: - TableView Delegate
extension ViewController: UITableViewDelegate , UITableViewDataSource {
    
    func numberOfSections(in tableView: UITableView) -> Int {
        
        return 1
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        
        return data.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        let cell = tableView.dequeueReusableCell(withIdentifier: "MainPageCell") as! MainPageCell
        cell.model = data[indexPath.row] 
        return cell
    }
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        
        return 118
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        
        let vc = DetailViewController()
        vc.model = data[indexPath.row]
        vc.data = data as! NSMutableArray;
        vc.currentIndex = indexPath.row
        DispatchQueue.main.async {
            self.present(vc, animated: true, completion: nil)
        }
    }
}

